crates/core/tedge_agent/src/agent.rs

```diff
@@ -170,7 +206,7 @@ impl Agent {
     }
 
     #[instrument(skip(self), name = "sm-agent")]
-    pub async fn start(&mut self, v1: bool) -> Result<(), anyhow::Error> {
+    pub async fn start(self, v1: bool) -> Result<(), anyhow::Error> {
```

Comment:
I agree and spent a while trying to understand myself what `true` and `false` mean here. I'm happy to change this, but I don't have the best context on exactly what this means. @albinsuresh @reubenmiller can either of you suggest descriptive names for the variants of this parameter to avoid this confusion? Equally I understand if we decide "this is a feature flag that is destined to be removed before the 1.0 release so we'll just live with it for now" (but the change is likely completely trivial to make if good names spring to mind).

Reply:
Yeah, it's just a temporary flag that we introduced recently, which would/should be removed with #2457 . So, just ignore that for now.